### ISSUE TITLE
Calling unbound function yields null return despite call working correctly in Fiddler

### ISSUE THREAD

**Issue by spsingletary** [object/Simple.OData.Client#172]
When executing the following statement, I am able to view a successful call to OData v2 service in Fiddler, however the return does not throw an exception or populate the response object with a value.

```
var response = await client.Unbound<SendUserIdStatus>()
                                 .Function("SndUsrIdByTaxIdNBuagId")
                                            .Set(new SendUserIdStatusCollection()
                                            {
                                                BuagId = request.ContractAccountNo,
                                                TaxId = request.TaxId,
                                            })
                                            .ExecuteAsSingleAsync(); 
```

**Comment by object** [comment]
Do you see in Fiddler that you are getting back the response? If so, can you share with me your service metadata file and the copy of that response?

**Comment by apngithub** [comment]
Hello. I have encountered what I believe to be the same issue. In my case I can confirm the response is visible in Fiddler, however the programmatic response object is not populated with anything. Could you tell me if this has been reproduced and if it is slated to be fixed in the near future? Thanks.

**Comment by object** [comment]
In order to reproduce this issue in addition to code example I need your service metadata and your entity type definitions. Then I should be able to research this issue.

**Comment by apngithub** [comment]
Hi. Thanks for the response. I'm attaching a simple example of an OData service that exposes an unbound function and a client that invokes it. In the client, the call to ExecuteAsSingleAsync seems to just return an object (unlike others that return a dictionary) and there doesn't seem to be any way to extract the response.
[SimpleODataClient.zip](https://github.com/object/Simple.OData.Client/files/115919/SimpleODataClient.zip)
[ODataActionService.zip](https://github.com/object/Simple.OData.Client/files/115920/ODataActionService.zip)

**Comment by apngithub** [comment]
Sorry, I forgot to mention in the above example I use an unbound function called "TestMovie", but I've noted the same behaviour for unbound functions also.

**Comment by apngithub** [comment]
Sorry, I forgot to mention in the above posted example I've used an unbound function called "TestMovie", but I've noted the same behaviour in unbound Actions also.

**Comment by object** [comment]
Thank you very much, this is very helpful. I hope I will be able to nail down the bug.

**Comment by object** [comment]
As a workaround, you can rewrite the statement like this:

await client.Unbound().Function("TestMovie").ExecuteAsScalarAsync<string>()

I will investigate the problem further. Thanks to your example, I managed to reproduce it.

**Comment by object** [comment]
I believe this is fixed now in 4.14.3. Please verify.

**Comment by apngithub** [comment]
Hi there, 

The behaviour has changed but it doesn't appear to be fixed. The unbound function in this case returns a payload that looks like this:

{
  "odata.metadata":"http://localhost:9999/odata/$metadata#Collection(Edm.String)",
  "value":[
    "TestMovie1","TestMovie2","TestMovie3"
  ]
}

If I call the unbound function using ExecuteAsSingleAsync, it returns a string object whose value is "TestMovie1" (i.e. it only returns the first value of the collection). 

If I call the unbound function using ExecuteAsArrayAsync then the function returns an plain object. The object appears to have three sub-objects according to the VS debugger, but it's not obvious to me how we're supposed to access them programmatically and again the objects don't appear to be populated with values.

{object[3]}
    [0]: {object}
    [1]: {object}
    [2]: {object}

I tried specifying string as the generic parameter type to ExecuteAsArrayAsync but encountered a System.AggregateException. The inner exception was: {"Unable to create an instance of type String that does not have a default constructor."} 

Perhaps there is some other method I'm supposed to be calling? The service returns a collection of strings, so it was our expectation that the returned object from the call would be an array or list type. Also, we have some functions and actions that return entities that are serialized as object graphs. In such cases, shouldn't these calls return dictionaries? Please let me know if we've misunderstood the usage of the client API.

Thanks again for your help.

**Comment by object** [comment]
Well, actually ExecuteAsSingle behaves now correctly. It is supposed to extract single (first) element of the result, and it does. What you are supposed to use if you want the whole collection is ExecuteAsArray<string> but apparently it is still broken and I am going to have a look at it.

**Comment by apngithub** [comment]
Thanks for the clarification about its usage. Could you please tell me what the appropriate usage is for more complex responses? We have some unbound actions and functions that return responses like the one posted below. We need to be able to extract the values of any and all members of the response object using typeless syntax. 

{
  "@odata.context": "http://localhost:5000/api/$metadata#Samples/$entity",
  "@odata.type": "#Sample.AddItemResponse",
  "Id": "100e4c87af404105b3c6b0532c164093",
  "ResponseCode": "Ok",
  "Messages": [],
  "Models": [
    {
      "@odata.type": "#Sample.ItemAdded",
      "Name": "",
      "ItemId": "4b43799956b04eec9c0b79eea30b9de5",
      "Included": true
    },
    {
      "@odata.type": "#Sample.UnitTypes",
      "Name": "",
      "UnitTypeA": {
        "Code": "123",
        "Amount": 18
      },
      "UnitTypeB": {
        "Code": "345",
        "Amount": 0
      },
      "UnitTypeC": {
        "Code": "678",
        "Amount": 18
      },
      "UnitTypeD": {
        "Code": "910",
        "Amount": 0
      }
    }
  ],
  "ActionUrl": null
}

**Comment by object** [comment]
AFAIK it should be sufficient to declare types that match your response and use either ExecuteAsArray (if you response is a collection) or ExecuteAsSingle  or Scalar if response is a single item or a scalar value.

**Comment by object** [comment]
Fixed in version 4.15.2.
